sources/web/datalab/polymer/test/item-list-test.ts

```diff
@@ -281,4 +311,113 @@ describe('<item-list>', () => {
     assert(!isSelected(0), 'first item should not be selected when selection is disabled');
   });
 
+  describe('inline details', () => {
+    it('should not create a details element for an item without details', () => {
+      testFixture.inlineDetailsMode = InlineDetailsDisplay.SINGLE_SELECT;
+      const firstRow = getRow(0);
+      const firstRowDetailsContainer = getRowDetailsContainer(0);
+      firstRow.click();
+
+      assert(!firstRowDetailsContainer.firstChild,
+          'first item should not show details when clicked');
+    });
+
+    it('should create a details element for an item with details', () => {
+      testFixture.inlineDetailsMode = InlineDetailsDisplay.SINGLE_SELECT;
+      const fourthRow = getRow(3);
+      const fourthRowDetailsContainer = getRowDetailsContainer(3);
+      fourthRow.click();
+
+      const detailsElement: HTMLElement =
+          fourthRowDetailsContainer.firstChild as HTMLElement;
+      assert(!!detailsElement,
+          'fourth item should create details element when clicked');
+      assert(fourthRowDetailsContainer.getAttribute('hidden') == null,
+          'fourth details container should be visible');
+    });
+
+    it('should not create a details element in NONE display mode', () => {
+      testFixture.inlineDetailsMode = InlineDetailsDisplay.NONE;
+      const fourthRow = getRow(3);
+      const fourthRowDetailsContainer = getRowDetailsContainer(0);
+      fourthRow.click();
+
+      assert(!fourthRowDetailsContainer.firstChild,
+          'fourth item should not create details element when clicked');
+      assert(fourthRowDetailsContainer.getAttribute('hidden') != null,
+          'fourth details container should not be visible');
+    });
+
+    it('should show details even when not clicked in ALL mode', () => {
+      testFixture.inlineDetailsMode = InlineDetailsDisplay.ALL;
+      // TODO: the current behavior of ALL does not show all of the details when
+      // the page first renders, but only displays the details when the
+      // selection state for an item changes. We might want to change this
+      // behavior so that it figures out in advance that it should show
+      // all the details.
+      testFixture._unselectAll();   // Recalculate details display for all items
+      const fourthRowDetailsContainer = getRowDetailsContainer(3);
```

Comment:
Should we make sure that all the items that can create a details element have one?